### PR TITLE
Update FairnessDashboard for named argument MetricFrame

### LINKED ISSUE #770: Keyword Arguments Change Affecting Fairness Dashboard
Originally opened in https://github.com/fairlearn/fairlearn/issues/933
@michaelamoako FYI

With the new change that arguments must be keywords, a warning is thrown when attempting to pass in multiple predictions into y_pred as shown in the sample code below. This seems to be an issue specifically in the dashboard.

Describe the bug
Key work argument requirement leading to bug in the Fairness dashboard, but not in the API

Steps/Code to Reproduce
```
FairnessDashboard(sensitive_features=sensitive_cols,
                y_true=golden_labels,
                y_pred=predictions)
```
Expected Results
No error is thrown.

Actual Results
You have provided 'metrics', 'y_true', 'y_pred' as positional arguments. Please pass them as keyword arguments. From version 0.10.0 passing them as positional arguments will result in an error.

Screenshots
image

### PR DESCRIPTION
**Pull request by riedgar-ms**
A recent update to Fairlearn has started the process of moving to requiring named arguments for `MetricFrame`. Update `FairnessDashboard` to match.

Closes #770

### CODE REVIEW

**Comment by riedgar-ms** [comment]
And I've run the two modified notebooks, and checked that the dashboard is no longer emitting warnings.

File: notebooks/fairness-dashboard-loan-allocation.ipynb [hunk 1]
```diff
@@ -302,24 +309,18 @@
   }
  ],
  "metadata": {
+  "interpreter": {
```
Comment (xuke444) [inline notebooks/fairness-dashboard-loan-allocation.ipynb:312]: Is this your local env setup?
Probably not good to check in.

Comment (riedgar-ms) [reply notebooks/fairness-dashboard-loan-allocation.ipynb]: Arghh, hadn't spotted that something 'helpfully' did that. Thanks. Will remove.

**Review by romanlutz — APPROVED** [review]
Thanks! Does this need updating in AzureML notebooks, too?

**Review by xuke444 — APPROVED** [review]

**Pull request merged**